compiler/src/dotty/tools/io/ZipArchive.scala

```diff
@@ -274,12 +206,11 @@ final class ManifestResources(val url: URL) extends ZipArchive(null) {
     for (zipEntry <- iter) {
       val dir = getDir(dirs, zipEntry)
       if (!zipEntry.isDirectory) {
-        class FileEntry() extends Entry(zipEntry.getName) {
+        val f = new Entry(zipEntry.getName) {
```

Comment:
I think this change is no longer necessary now that https://github.com/lampepfl/dotty/pull/2282 got merged.